### ISSUE TITLE
UnlabeledSMILES fails with pre_train node/edge featurizer

### ISSUE THREAD

**Issue by skrsna** [awslabs/dgl-lifesci#65]
Hi, 

I noticed that `UnlabeledSMILES` class in `regression_inference.py` fails if one uses checkpoints from any GIN models. The specific error I got is: 

<details>
<summary>Error Traceback.</summary>
<p>

```
Traceback (most recent call last):
  File "/home/ubuntu/Code/dgl-lifesci/examples/property_prediction/csv_data_configuration/regression_inference.py", line 104, in <module>
    main(args)
  File "/home/ubuntu/Code/dgl-lifesci/examples/property_prediction/csv_data_configuration/regression_inference.py", line 19, in main
    edge_featurizer=args['edge_featurizer'])
  File "/home/ubuntu/excape/lib/python3.7/site-packages/dgllife/data/smiles_inference.py", line 55, in __init__
    edge_featurizer=edge_featurizer))
  File "/home/ubuntu/excape/lib/python3.7/site-packages/dgllife/utils/mol_to_graph.py", line 225, in mol_to_bigraph
    canonical_atom_order, explicit_hydrogens)
  File "/home/ubuntu/excape/lib/python3.7/site-packages/dgllife/utils/mol_to_graph.py", line 79, in mol_to_graph
    g.edata.update(edge_featurizer(mol))
  File "/home/ubuntu/excape/lib/python3.7/_collections_abc.py", line 841, in update
    self[key] = other[key]
  File "/home/ubuntu/excape/lib/python3.7/site-packages/dgl/view.py", line 133, in __setitem__
    self._graph.set_e_repr({key : val}, self._edges)
  File "/home/ubuntu/excape/lib/python3.7/site-packages/dgl/graph.py", line 2373, in set_e_repr
    ' Got %d and %d instead.' % (nfeats, num_edges))
dgl._ffi.base.DGLError: Expect number of features to match number of edges. Got 87 and 60 instead.
```
</details>

This can be solved by adding `self_loop=True` to `mol_to_graph` function in `UnlabeledSMILES` class. I can open a PR with the fix. Thanks for this awesome tool.

**Comment by mufeili** [comment]
Thank you for the kind words and nice catch! In that case, I would recommend 

```python
from dgllife.utils import mol_to_bigraph
from functools import partial

if args['model'] in ['gin_supervised_contextpred', 'gin_supervised_infomax', 
                     'gin_supervised_edgepred', 'gin_supervised_masking']:
    from dgllife.utils import PretrainAtomFeaturizer, PretrainBondFeaturizer
    args['atom_featurizer_type'] = 'pre_train'
    args['bond_featurizer_type'] = 'pre_train'
    args['node_featurizer'] = PretrainAtomFeaturizer()
    args['edge_featurizer'] = PretrainBondFeaturizer()
    args['mol_to_bigraph'] = partial(mol_to_bigraph, add_self_loop=True)
else:
    args['mol_to_bigraph'] = mol_to_bigraph
````

Also I doubt this is the same case for `classification_inference.py`. You may go ahead to open a PR for the fix. Thank you.

**Comment by mufeili** [comment]
Fixed by @skrsna in #66 .
